tiledb/sm/storage_manager/storage_manager.cc

```diff
@@ -1504,8 +1510,10 @@ void StorageManager::increment_in_progress() {
 }
 
 Status StorageManager::is_array(const URI& uri, bool* is_array) const {
-  RETURN_NOT_OK(
-      vfs_->is_file(uri.join_path(constants::array_schema_filename), is_array));
+  URI schema_uri;
+  RETURN_NOT_OK(get_latest_array_schema_uri(uri, &schema_uri));
```

Comment:
This will be expensive. I think we need to check for existence of folder `<array_uri>/__meta` instead.

Reply:
I am minimizing the use of is_file, is_dir of vfs.

Reply:
This line will propagate to a `VFS::ls`, which is more expensive than `is_dir`.

Reply:
changed to check if the <array_uri>/__schema exists or not. If not exist, we will check further if <array_uri>/__array_schema.tdb exists or not